### ISSUE TITLE
Revamp code dealing with _tui_terminal_out

### ISSUE THREAD

**Issue by jeffwright13** [jeffwright13/pytest-tui#66]
There is a global (`_tui_terminal_out`) that is used to gather ansi-encoded info and send it to file. However, if pytest is invoked without --tui using only the --version flag, an error occurs:
```
~/coding/pytest-tui on  main! ⌚ 22:58:19
$  cd /Users/jwr003/coding/pytest-tui ; /usr/bin/env /Users/jwr003/coding/pytest-tui/venv/bin/python /Users/jwr003/.vscode/extensions/ms-python.python-2022.
14.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 61620 -- /Users/jwr003/coding/pytest-tui/venv/bin/pytest --version 
pytest 7.2.0.dev264+g6ad32a9c5
sys:1: ResourceWarning: unclosed file <_io.BufferedRandom name=4>
```

We need that global because some processing code (e.g. `pytest_runtest_logstart`) is required that does not give access to the Pytest Config object.

However, it may be that the global doesn't have to be instantiated (only declared) in the global scope, which would simplify the use of this variable. As it is now we have a bunch of independent code all closing that tempfile out so that the pytest session doesn't end without it being closed.

**Comment by jeffwright13** [comment]
See issue #67

**Comment by jeffwright13** [comment]
Took a bit of doing but fixed in 1.4.2.
